pkg/apis/workspaces/v1alpha2/endpoint.go

```diff
@@ -47,6 +47,8 @@ const (
 )
 
 type Endpoint struct {
+	// +kubebuilder:validation:Pattern=^([a-z0-9]([-a-z0-9]*[a-z0-9]))?$
+	// +kubebuilder:validation:MaxLength=63
```

Comment:
Should we have a shorted maxlength for endpoints? I believe k8s treats some routing objects (e.g. port names) differently (DNS subdomain, e.g. [here](https://github.com/kubernetes/kubernetes/blob/53a9d106c4aabcd550cc32ae4e8004f32fb0ae7b/pkg/util/validation.go#L54-L64))

Reply:
The code you pointed to was for DNS952. Do we need to follow the rules for DNS 952?

Reply:
I'm not sure, and it's probably fine to use 63 chars for now. We just may want to keep in mind in the future that on the Kubernetes end, some things are more restricted.

Reply:
In kube doc, it states the maxlength for name/id is 63 (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/), which follows DNS1123. 
But for the regex, it is different, even the doc and regex states the name can start and end with an alphanumeric character, but when I manually tested, a numeric name is not valid.

Reply:
Our main goal is to have the fields to be able to easily translate into K8 resources with minimal impact on other environments. A char limit of 24 looks fairly limiting and users can easily go beyond that. I'd suggest using the DNS 952 regex and allow 63 chars limited.

Reply:
Yeah I've found kube documentation to be a little lacking here. I think the current system is good for now and we can potentially refine it if need be later.